arm/Microsoft.OperationalInsights/workspaces/deploy.bicep

```diff
@@ -0,0 +1,844 @@
+  name: 'sampleIISLog1'
+  kind: 'IISLogs'
+  properties: {
+    state: 'OnPremiseEnabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleSyslog1 'datasources@2020-03-01-preview' = {
+  name: 'sampleSyslog1'
+  kind: 'LinuxSyslog'
+  properties: {
+    syslogName: 'kern'
+    syslogSeverities: [
+      {
+        severity: 'emerg'
+      }
+      {
+        severity: 'alert'
+      }
+      {
+        severity: 'crit'
+      }
+      {
+        severity: 'err'
+      }
+      {
+        severity: 'warning'
+      }
+    ]
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleSyslogCollection1 'datasources@2020-03-01-preview' = {
+  name: 'sampleSyslogCollection1'
+  kind: 'LinuxSyslogCollection'
+  properties: {
+    state: 'Enabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleLinuxPerf1 'datasources@2020-03-01-preview' = {
+  name: 'sampleLinuxPerf1'
+  kind: 'LinuxPerformanceObject'
+  properties: {
+    performanceCounters: [
+      {
+        counterName: '% Used Inodes'
+      }
+      {
+        counterName: 'Free Megabytes'
+      }
+      {
+        counterName: '% Used Space'
+      }
+      {
+        counterName: 'Disk Transfers/sec'
+      }
+      {
+        counterName: 'Disk Reads/sec'
+      }
+      {
+        counterName: 'Disk Writes/sec'
+      }
+    ]
+    objectName: 'Logical Disk'
+    instanceName: '*'
+    intervalSeconds: 10
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleLinuxPerfCollection1 'datasources@2020-03-01-preview' = {
+  name: 'sampleLinuxPerfCollection1'
+  kind: 'LinuxPerformanceCollection'
+  properties: {
+    state: 'Enabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_AzureActivityLog 'datasources@2020-03-01-preview' = [for item in activityLogAdditionalSubscriptionIDs: {
+  kind: 'AzureActivityLog'
+  name: '${(empty(activityLogAdditionalSubscriptionIDs) ? 'placeholder' : item)}'
+  properties: {
+    linkedResourceId: '/subscriptions/${item}/providers/microsoft.insights/eventTypes/management'
+  }
+  dependsOn: [
+    logAnalyticsWorkspace
+  ]
+}]
+
+resource storageinsightconfigs 'storageinsightconfigs@2020-03-01-preview' = if (!empty(diagnosticStorageAccountId)) {
+  name: '${diagnosticStorageAccountName}'
+  properties: {
+    containers: []
+    tables: [
+      'WADWindowsEventLogsTable'
+      'WADETWEventTable'
+      'WADServiceFabric*EventTable'
+      'LinuxsyslogVer2v0'
+    ]
+    storageAccount: {
+      id: diagnosticStorageAccountId
+      key: (empty(diagnosticStorageAccountId) ? '' : listKeys(diagnosticStorageAccountId, '2016-12-01').keys[0].value)
+    }
+  }
+}
+
+resource linkedServices 'linkedServices@2020-03-01-preview' = if (!empty(automationAccountId)) {
+  name: 'Automation'
+  properties: {
+    resourceId: automationAccountId
+  }
+}
+}
+
+@batchSize(1) // Serial loop deployment
+resource solutionRes 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = [for solution in solutions: if (!empty(solutions)) {
```

Comment:
I'd suggest `solutions`, unless creating conflicts with the solutions parameter

Reply:
It does. That's the only reason I used it. I guess the alternative would be the rename the parameter to e.g. `solutionsDef` os similar

Reply:
updated the parameter